# 仿真疑难杂症论

by zhangao

本文件用于解决仿真中遇到的各种bug

##### 1.AttributeError: 'int' object has no attribute '__rdiv__'

![image-20240626210250732](C:\Users\lenovo\AppData\Roaming\Typora\typora-user-images\image-20240626210250732.png)

解决方式：将提示错误所在的地方的函数改成异步函数

![image-20240626210935744](C:\Users\lenovo\AppData\Roaming\Typora\typora-user-images\image-20240626210935744.png)

##### 2. 使用python向模块发送一个输入测试序列时，波形图显示输入握手信号全0

##### ![image-20240627112233776](C:\Users\lenovo\AppData\Roaming\Typora\typora-user-images\image-20240627112233776.png)

原因在于，在输入一个序列的数据的时候，python仅仅是将数据发出去，送到python的发送序列中，发完就结束了，python在这个发送循环之外没有写其他逻辑的话，仿真在发完数据之后就结束了，没有等到verilog对输入进行处理（verilog和python是异步执行的），就认为仿真结束了，生成的波形文件里面就没有记录到模块信号。

![image-20240627112312416](C:\Users\lenovo\AppData\Roaming\Typora\typora-user-images\image-20240627112312416.png)

解决方式是：

1. 在输入完数据之后强行等待一段时间，等verilog处理完毕再结束仿真

   ![image-20240627112847695](C:\Users\lenovo\AppData\Roaming\Typora\typora-user-images\image-20240627112847695.png)

2. 在python里写：收到verilog对应的响应之后再结束仿真

